update_factorio.py

```diff
@@ -138,11 +138,28 @@ def verbose_aware_exec(exec_args, verbose=False):
         raise
 
 
+def get_username_token(factoriopath):
+    with open(os.path.normpath(factoriopath + '/../../../player-data.json')) as data_file:  
```

Comment:
Thanks, I hadn't noticed that! That's very good! But it's reliant on having the login credentials in `server-settings.json`, which is not required (for a long time my server didn't do that, I only recently turned it on so I could use whitelist instead of password).

And the `player-data.json` still might be living elsewhere; I've tracked it down in depth and will make a top-level comment with the details.